### ISSUE TITLE
Separating sleep and activity monitoring from GadgetBridge

### ISSUE THREAD

**Issue by Avamander** [Freeyourgadget/Gadgetbridge#534]
As the title already says, I think those functions should be separated from GadgetBridge and instead something named like GadgetHealth could be installed that allows access to that data.

I am sure it would benefit both the developers and users by allowing mainly three things:
1. 3rd party apps could use the data through the same API as GadgetHealth (providing alternatives)
2. GadgetBridge and GadgetHealth can updated and developed independently.
3. People don't have to install GadgetHealth if they do not wish. If they do, people actually have a nice-to-use health application they can show off to their friends if they wish.

Basically making GadgetBridge follow the "do one thing and do it good" moto (being a bridge not a fitness/sleep app).

**Comment by bluuub** [comment]
I think this is very user specific. In my case that is exactly what I want. 
Having one app to control all features (like setting alarms, checking my sleep data, checking daily steps, ...) is perfect for me. Having to switch between several apps (with options and settings and so on) would make my life harder.

**Comment by Avamander** [comment]
Though why does GadgetBridge deal with alarms? There's Alarm/Clock app preinstalled on all phones for that. That functionality should maybe too be left for those apps and just sync the alarms or vibrate if needed.

**Comment by Avamander** [comment]
I find that there are apps in all categories you named that could do a lot more with the data than GadgetBridge could ever do just based on the fact that they specialize in those aspects. Though I see the appeal in simple apps that take the data and display it simply and quickly and thus providing GadgetHealth and unbundling those features from GadgetBridge would allow anyone to just keep those simple features.

**Comment by 6arms1leg** [comment]
Sorry, but I also vote against this issue. I prefer GB the way it is now.
While being an advocate of modularity and *nix principles, I'm not sure if the mentioned Unix philosophy should be applied to Android apps.

> [...] by all the other alarm applications on GPlay.

Some people here (+me) don't have Google on their phone.

**Comment by IzzySoft** [comment]
> Though why does GadgetBridge deal with alarms? There's Alarm/Clock app preinstalled on all phones for that.

And that synchronizes with your Pebble/Mi:Band/ZeBand/… without any help? I doubt that. I fully agree with @6arms1leg and @bluuub here: While I'm not a big fan of feature-overladen software, GB definitely does *not* fall into that category. Admitted, the health graphs are quite basic (fine with me) – but nobody keeps you from using a 3rd party app for that, GB even supports it.

**Comment by Avamander** [comment]
Google Play or FDroid, there's more than one Alarm app, GB should deal with
synchronization of wearable data, bridging the two systems. (Also, yes,
Sleep as Android interacts with Pebble)

If you are fine with the functionality there is at the moment that's good,
though the separation most likely wouldn't affect that. Providing a clear
cut between the completely different functionalities ensures future
compatibility of the *nix philosophy.

**Comment by bluuub** [comment]
Meaning I'd have to install a bloated version of every app that does anything with my device? the other option would require system level integration of it (which is not going to happen and cannot be fixed by installing an app, but requires messing with custom android builds). While that works perfectly on linux, android apps do not work like that. Every app has it's own little sandbox, I cannot pipe any say alarm data from my clock app through a synchronization app for my device. Hence clock, health, activity, ... every app would have to be able to talk to x wearables while every single one of them has another API. Bloat bloat bloat. 
I do prefer having that "one app that does it all" for a given functionality. 
It would be a nightmare to make sure every app is connected to my Mi Band.

**Comment by danielegobbetti** [comment]
https://lwn.net/Articles/710930/

s/Gentoo/Gadgetbridge/
> Ultimately Gadgetbridge is working together to produce a product.  If it is
a product that you want to use, then use it.  If it is a product that
you don't want to use, please contact us for a full refund, and
consider contributing to somebody who makes a product that you do want
to use. 

closing and locking this
